src/countdown-timer/countdown-timer.tsx

```diff
@@ -0,0 +1,154 @@
+    });
+
+    const isMobile = useMediaQuery({
+        maxWidth: MediaWidths.mobileL,
+    });
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+
+    useEffect(() => {
+        setIsPlaying(show);
+    }, [show]);
+
+    useEffect(() => {
+        if (timer === 0) {
+            performOnFinishHandler();
+        } else if (timer === notifyTimer) {
+            performOnDurationHandler();
+            performOnNotifyHandler();
+        } else if (timer < notifyTimer) {
+            if (!isNotified.current) performOnNotifyHandler();
+            performOnDurationHandler();
+        }
+    }, [timer]);
+
+    useEffect(() => {
+        const y = getOffsetY();
+        setOffsetY(y);
+    }, [isMobile, wrapperRef.current]);
+
+    // =============================================================================
+    // HELPER FUNCTIONS
+    // =============================================================================
+
+    const performOnDurationHandler = () => {
+        if (onDuration) {
+            const { seconds } = TimeHelper.convertSecondsToTime(timer);
+
+            onDuration(seconds);
+        }
+    };
+
+    const performOnNotifyHandler = () => {
+        if (!isNotified.current) isNotified.current = true;
+
+        if (onNotify) onNotify();
+    };
+
+    const performOnFinishHandler = () => {
+        setIsPlaying(false);
+
+        if (onFinish) onFinish();
+    };
+
+    function getOffsetY() {
+        const desktopTop = offset?.top ?? 168;
+        const mobileTop = mobileOffset?.top ?? 80;
+
+        const offsetY = !isMobile ? desktopTop : mobileTop;
+
+        return offsetY;
+    }
+
+    // =============================================================================
+    // RENDER FUNCTION
+    // =============================================================================
+
+    const renderCountdown = (position: Position) => {
+        const {
+            hours,
+            minutes: _minutes,
+            seconds,
+        } = TimeHelper.convertSecondsToTime(timer);
+        const clientRect = wrapperRef.current?.getBoundingClientRect();
+
+        const minutes = hours * 60 + _minutes;
+        const m = minutes !== 1 ? "mins" : "min";
+        const s = seconds !== 1 ? "secs" : "sec";
+        const offsetX = !isMobile ? offset?.left : mobileOffset?.left;
+
+        return (
+            <Countdown
+                data-testid={testId}
+                data-id={
+                    position === "relative"
+                        ? `countdown-wrapper`
+                        : `fixed-countdown-wrapper`
+                }
+                ref={position === "relative" ? wrapperRef : undefined}
+                inert={onScreen ? undefined : ""}
+                $isFixed={position === "fixed"}
+                $pinned={position === "fixed" && !onScreen}
```

Comment:
what's the difference between isFixed and pinned? could `position: fixed` be declared in the pinned style instead?